app/src/main/java/net/opendasharchive/openarchive/services/webdav/WebDAVSiteController.kt

```diff
@@ -418,9 +418,13 @@ class WebDAVSiteController : SiteController {
             fos.close()
             sardine?.put(urlMeta, fileMetaData, "text/plain", false, null)
             if (getUseProofMode()) {
+                var metaMediaHash = ProofMode.generateProof(mContext, Uri.parse(media.originalFilePath),media.mediaHashString)
```

Comment:
Why is this code doubled? 
These classes have a common ancestor: `SiteController`. Can you please refactor into that? No code duplication.

Reply:
done!